### PR TITLE
Create Radix Sort

### PR DESCRIPTION
**Pull request by Nyxoy77**
An efficient method of sorting

### CODE REVIEW

File: Know Your Code/C++/Radix Sort [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    for (int i = 1; i < arr.size(); i++) {
+        if (arr[i] > max) {
+            max = arr[i];
+        }
+    }
+    return max;
+}
+
+// Using counting sort to sort the elements based on significant places
+void countingSort(vector<int>& arr, int exp) {
```
Comment (AishwaryJain07) [inline Know Your Code/C++/Radix Sort:18]: void countingSort(vector<int>& arr, const int exp) {

Comment (AishwaryJain07) [reply Know Your Code/C++/Radix Sort:18]: Use const for read-only variables

**Review by AishwaryJain07 — CHANGES_REQUESTED** [review]
`#include <iostream>
#include <vector>

using namespace std;

// Function to find the maximum number to determine the number of digits
int getMax(const vector<int>& arr) {
    int max = arr[0];
    for (const int& value : arr) {
        if (value > max) {
            max = value;
        }
    }
    return max;
}

// Using counting sort to sort the elements based on significant places
void countingSort(vector<int>& arr, const int exp) {
    const int n = arr.size();
    vector<int> output(n);
    vector<int> count(10, 0);

    // Count the occurrences of each digit
    for (const int& value : arr) {
        count[(value / exp) % 10]++;
    }

    // Update count[i] to store the position of the next occurrence of the digit
    for (int i = 1; i < 10; i++) {
        count[i] += count[i - 1];
    }

    // Build the output array
    for (int i = n - 1; i >= 0; i--) {
        output[count[(arr[i] / exp) % 10] - 1] = arr[i];
        count[(arr[i] / exp) % 10]--;
    }

    // Copy the output array to the original array
    for (int i = 0; i < n; i++) {
        arr[i] = output[i];
    }
}

// Radix Sort function
void radixSort(vector<int>& arr) {
    int max = getMax(arr);

    // Perform counting sort for every digit
    for (int exp = 1; max / exp > 0; exp *= 10) {
        countingSort(arr, exp);
    }
}

// Function to print an array
void printArray(const vector<int>& arr) {
    for (const int& value : arr) {
        cout << value << " ";
    }
    cout << endl;
}

int main() {
    vector<int> arr = {170, 45, 75, 90, 802, 24, 2, 66};
    cout << "Original array: ";
    printArray(arr);

    radixSort(arr);

    cout << "Sorted array: ";
    printArray(arr);

    return 0;
}
`

File: Know Your Code/C++/Radix Sort (hunk 1 above)
Comment (Nyxoy77) [reply Know Your Code/C++/Radix Sort:18]: Okay

**Review by AishwaryJain07 — APPROVED** [review]

**Pull request merged**